src/main/java/bowling/pin/Pin.java

```diff
@@ -0,0 +1,65 @@
+package bowling.pin;
+
+import java.util.Objects;
+
+public class Pin {
+
+  private static final int MAX_PINS = 10;
+  private static final int MIN_PINS = 0;
+
+  private static final String MSG_ERROR_PIN_MAX = "입력받은 핀의 수가 %d보다 클 수 없습니다.";
+  private static final String MSG_ERROR_PIN_MIN = "입력받은 핀의 수가 %d보다 작을 수 없습니다.";
+  private static final String MSG_ERROR_DOWN_PIN_MAX = "쓰러트린 핀의 총합이 %d보다 클 수 없습니다.";
+
+  private final int fallenPin;
+
+  private Pin(final int fallenPin) {
+    this.fallenPin = fallenPin;
+  }
+
+  public static Pin from(final int fallenPin) {
+    validationPins(fallenPin);
+    return new Pin(fallenPin);
+  }
+
+  private static void validationPins(final int fallenPin) {
+    if (fallenPin > MAX_PINS) {
+      throw new IllegalArgumentException(String.format(MSG_ERROR_PIN_MAX, MAX_PINS));
+    }
+
+    if (fallenPin < MIN_PINS) {
+      throw new IllegalArgumentException(String.format(MSG_ERROR_PIN_MIN, MIN_PINS));
+    }
+  }
+
+  public int totalDownPin(final Pin firstPin) {
```

Comment:
```suggestion
  public Pin totalDownPin(final Pin firstPin) {
```

Pin에서 제공하는 오퍼레이션에선 최대한 원시값이 아닌 Pin 클래스를 활용해보아요 ㅎㅎ

Reply:
객체를 반환하게 되면, 해당 원시값을 사용하는 부분을 전부 변경해야 하는데요.
값자체를 필요로 하는 상황에 대해서는 dto를 쓰는 수 밖에 없는건가요?
원시값을 사용하는게 무조건 안 좋은 방향인지 좀 궁금합니다.

원시값이 필요하다는 얘기는 getter를 의미하는데 
이는 책임에 대한 문제로 이어지고 getter를 쓰는 내용을 보통
책임 도메인에서 원하는 결과값을 메세지로 반환하는 방식을 통해서 getter를 제거하고 있는데요.

이 부분은 참 제 입장에서는 좀 애매한 부분이 있습니다.
현재 다른 사용처는 객체 반환을 변경해도 별 다른 리팩토링 리소스가 들지 않습니다.
문제는 파이널 프래임에서 2구를 굴려서 10개의 핀이 쓰러졌는지를 확인하는 부분이 이제 좀 마음에 걸려요.
핵심은 10개가 쓰러졌는가를 알아야하는 부분이니 이 부분이 객체 비교로 잘 대체가 되는지 확인을 해봐야 할 것 같습니다.

Reply:
객체로 비교하는 걸로 잘 교체가 되이었습니다.
다만 의문은 좀 남네요.
기존의 원시값 int형을 쓰면 장점이 null에 대한 대응을 별도로 구현안하고도 코드가 수행이 가능했었지만
객체로 비교하게 함으로 인해 null이 발생할 수 있는 조건이 활성화되었고
이를 해결하기 위해 optoinal을 사용해서 대응을 했습니다.

코드자체로는 객체를 비교하는 것으로 인해 개념적으론 좋아보이지만.
이러한 코드를 위해 비용이 추가된 부분을 놓고 고민을 해보았습니다.
이렇게 하는게 더 좋은 코드인것 같기도 하고 불필요한 오버엔지니어링이 아닌지 의심이 가기도 합니다.
객체지향적으로 보면 분명 더 나은 코드이긴 할 것이지만, 굳이 이렇게까지 해야 하나를 아직 고민하는 걸 보면
아직 객체지향 마인드가 부족해서 인가 싶기도 합니다.
보이는 만큼 생각이 드네요. 오버엔지니어링을 경계하는 것도 고민하는 입장에서는 이런게 참 어렵습니다.